### DISCUSSION TITLE
Should runtime requirements be a bitflags instead of HashSet?

### DISCUSSION THREAD

**Discussion by HerringtonDarkholme** [web-infra-dev/rspack#2452] [Ideas]
Most of runtime_requirements store a set of static str, which is known beforehand.

Ideally we can use bitflags to reduce some memory allocation for it.

**Comment by IWANABETHATGUY** [comment]
Agree.

**Comment by hyf0** [comment]
Closed by https://github.com/web-infra-dev/rspack/pull/2461
